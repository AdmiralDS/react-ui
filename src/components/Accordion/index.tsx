import * as React from 'react';
import styled from 'styled-components';
import { typography } from '#src/components/Typography';
import { ReactComponent as ChevronDownOutline } from '@admiral-ds/icons/build/system/ChevronDownOutline.svg';
import { uid } from '#src/components/common/uid';
import { keyboardKey } from '#src/components/common/keyboardKey';

import { Collapse } from './Collapse';
import { moveFocus, nextItem, previousItem } from './utils';

type Dimension = 'l' | 'm';

const Chevron = styled(ChevronDownOutline)`
  transition: all 0.3s;
  flex-shrink: 0;
  width: 24px;
  height: 24px;
  margin-left: 8px;
  & *[fill^='#'] {
    fill: ${({ theme }) => theme.color.text.secondary};
  }
  [data-disabled='true'] & {
    & *[fill^='#'] {
      fill: ${({ theme }) => theme.color.text.tertiary};
    }
  }
  [data-dimension='m'] & {
    width: 20px;
    height: 20px;
  }
`;

const ItemTitleContent = styled.span`
  position: relative;
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: space-between;
  align-items: flex-start;

  padding: 16px 16px 15px 16px;
  [data-dimension='m'] && {
    padding: 10px 16px 9px 16px;
  }
`;

const ItemTitle = styled.button`
  position: relative;
  display: flex;
  justify-content: space-between;
  align-items: center;
  min-height: 54px;
  width: 100%;
  background: transparent;
  -webkit-tap-highlight-color: transparent;
  text-align: left;
  border: none;
  background: none;
  margin: 0;
  padding: 0;
  cursor: pointer;
  overflow: visible;
  ${typography['Subtitle/Subtitle 2']}
  &:hover {
    background: ${(p) => p.theme.color.background.secondary};
    & *[fill^='#'] {
      fill: ${(p) => p.theme.color.basic.hover};
    }
  }
  &:focus {
    &:before {
      position: absolute;
      content: '';
      border: 2px solid ${({ theme }) => theme.color.basic.hover};
      top: -1px;
      left: 0;
      bottom: -1px;
      right: 0;
    }
  }
  &:focus,
  & > ${ItemTitleContent} {
    outline: none;
  }

  [data-dimension='m'] & {
    min-height: 38px;
    ${typography['Subtitle/Subtitle 3']}
  }
`;

const ItemWrapper = styled.div<{ opened?: boolean; disabled?: boolean }>`
  border-bottom: 1px solid ${(p) => p.theme.color.background.tertiary};
  color: ${(p) => p.theme.color.text.primary};
  & > ${ItemTitle} ${Chevron} {
    transform: ${(p) => (p.opened ? 'rotate(180deg)' : 'rotate(0deg)')};
  }
  ${({ disabled }) => disabled && 'pointer-events: none;'}
`;

const ItemContent = styled.div<{ contentMaxHeight: number | string }>`
  box-sizing: border-box;
  overflow-y: auto;
  max-height: ${(p) => p.contentMaxHeight};
  padding: 4px 16px 16px 16px;
  ${typography['Body/Body 1 Long']}
  [data-dimension='m'] & {
    padding: 4px 16px 10px 16px;
    ${typography['Body/Body 2 Long']}
  }
`;

const AccordionWrapper = styled.div<{ dimension?: Dimension }>`
  position: relative;
  & > ${ItemWrapper}:first-child {
    & ${ItemTitleContent} {
      padding: ${({ dimension }) => (dimension === 'l' ? '15px 16px' : '9px 16px')};
    }
    border-top: 1px solid ${({ theme }) => theme.color.background.tertiary};
  }
`;

export interface AccordionItemProps extends Omit<React.HTMLAttributes<HTMLButtonElement>, 'onClick' | 'title'> {
  /** Заголовок компонента */
  title: React.ReactNode;
  /** дефолтное (изначальное) состояние компонента (раскрыт/свернут) при неконтролируемом режиме работы */
  defaultExpanded?: boolean;
  /** состояние компонента (раскрыт/свернут) при контролируемом режиме работы */
  expanded?: boolean;
  /** Колбек на клик по компоненту */
  onClick?: (title: React.ReactNode, expanded: boolean, event: React.MouseEvent<HTMLButtonElement>) => void;
  /**  Устанавливает максимальную высоту, на которую открывается контент до появления вертикального скрола */
  contentMaxHeight?: number | string;
  /** Отключение компонента */
  disabled?: boolean;
}

export const AccordionItem: React.FC<AccordionItemProps> = ({
  children,
  title,
  id,
  defaultExpanded,
  expanded: userExpanded,
  onClick,
  contentMaxHeight = '100vh',
  disabled,
  ...props
}) => {
  const [expanded, setExpanded] = React.useState(defaultExpanded);
  const collapseOpened = userExpanded === undefined ? expanded : userExpanded;

  const itemId = React.useMemo(() => id || uid(), [id]);
  const ITEM_TITLE_ID = `accordion_title_${itemId}`;
  const ITEM_CONTENT_ID = `accordion_content_${itemId}`;

  const handleClick = React.useCallback(
    (event: React.MouseEvent<HTMLButtonElement>) => {
      if (userExpanded === undefined) {
        onClick?.(title, !expanded, event);
        setExpanded(!expanded);
      } else {
        onClick?.(title, !userExpanded, event);
      }
    },
    [expanded, userExpanded, onClick, title],
  );
  return (
    <ItemWrapper opened={collapseOpened} data-disabled={disabled} disabled={disabled}>
      <ItemTitle
        onClick={disabled ? undefined : handleClick}
        role="button"
        type="button"
        aria-expanded={collapseOpened}
        aria-controls={ITEM_CONTENT_ID}
        id={ITEM_TITLE_ID}
        {...props}
      >
        <ItemTitleContent tabIndex={-1}>
          {title}
          <Chevron aria-hidden />
        </ItemTitleContent>
      </ItemTitle>
      <Collapse opened={collapseOpened} contentMaxHeight={contentMaxHeight}>
        <ItemContent
          contentMaxHeight={contentMaxHeight}
          role="region"
          aria-labelledby={ITEM_TITLE_ID}
          id={ITEM_CONTENT_ID}
        >
          {children}
        </ItemContent>
      </Collapse>
    </ItemWrapper>
  );
};

export interface AccordionProps extends React.HTMLAttributes<HTMLDivElement> {
  dimension?: Dimension;
}

export const Accordion: React.FC<AccordionProps> = ({ children, dimension = 'l', onKeyDown, ...props }) => {
  const accordionRef = React.useRef<HTMLDivElement | null>(null);
  const handleKeyDown = React.useCallback(
    (e) => {
      let focusedOption: any = ((accordionRef.current && accordionRef.current.ownerDocument) || document).activeElement;
      // если фокус находится на AccordionItem Title, то можно переключаться между accordion items с помощью ArrowDown, ArrowUp, Home, End
      if (focusedOption.id.indexOf('accordion_title') > -1 && focusedOption.tagName.toLowerCase() === 'button') {
        do {
          focusedOption = focusedOption?.parentNode;
        } while (focusedOption.parentNode !== accordionRef.current);

        const code = keyboardKey.getCode(e);

        if (code === keyboardKey.ArrowDown) {
          moveFocus(accordionRef.current, focusedOption, nextItem);
          e.preventDefault();
        } else if (code === keyboardKey.ArrowUp) {
          moveFocus(accordionRef.current, focusedOption, previousItem);
          e.preventDefault();
        } else if (code === keyboardKey.Home) {
          e.preventDefault();
          moveFocus(accordionRef.current, null, nextItem);
        } else if (code === keyboardKey.End) {
          e.preventDefault();
          moveFocus(accordionRef.current, null, previousItem);
        }
      }
      onKeyDown?.(e);
    },
    [accordionRef.current],
  );
  return (
    <AccordionWrapper
      ref={accordionRef}
      data-dimension={dimension}
      dimension={dimension}
      onKeyDown={handleKeyDown}
      {...props}
    >
      {children}
    </AccordionWrapper>
  );
};

Accordion.displayName = 'Accordion';
