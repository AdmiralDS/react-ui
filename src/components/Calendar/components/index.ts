export * from './Years';
export * from './Button';
export * from './Day';
export * from './DayNames';
export * from './Month';
export * from './Months';
export * from './Panel';
export * from './Week';
