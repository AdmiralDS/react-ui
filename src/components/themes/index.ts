export * from './light';
export * from './dark';
export type { Theme } from './common';
