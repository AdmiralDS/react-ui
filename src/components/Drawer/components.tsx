import type { FC, HTMLAttributes } from 'react';
import { createContext, useContext, useLayoutEffect, useRef } from 'react';
import styled from 'styled-components';

import { typography } from '#src/components/Typography';
import { debounce } from '#src/components/common/utils/debounce';
import { resizePaddings } from '../common/dom/resizePaddings';

const Title = styled.h5<{ $mobile: boolean; $displayCloseIcon: boolean }>`
  ${({ $mobile }) => ($mobile ? typography['Header/H6'] : typography['Header/H5'])};
  color: var(--admiral-color-Neutral_Neutral90, ${(p) => p.theme.color['Neutral/Neutral 90']});
  margin: 0;
  padding: ${({ $mobile, $displayCloseIcon }) => {
    if ($mobile) {
      return $displayCloseIcon ? '0 48px 8px 16px' : '0 16px 8px';
    }
    return $displayCloseIcon ? '0 56px 8px 24px' : '0 24px 8px';
  }};
`;

const Content = styled.div<{ $mobile: boolean }>`
  overflow-y: auto;
  outline: none;
  scrollbar-gutter: stable;
  padding-block: 8px;
  padding-inline: ${({ $mobile }) => `${$mobile ? 16 : 24}px`};
  height: 100%;
`;

const ButtonPanel = styled.div<{ $mobile: boolean }>`
  display: flex;
  flex-direction: ${({ $mobile }) => ($mobile ? 'column-reverse' : 'row-reverse')};
  padding: ${({ $mobile }) => ($mobile ? '16px 16px 0' : '16px 24px 0')};

  & > button {
    margin: ${({ $mobile }) => ($mobile ? '0 0 16px 0' : '0 16px 0 0')};
    ${({ $mobile }) => $mobile && 'width: 100%;'}
  }

  & > button:first-child {
    margin: 0;
  }
`;

export const DrawerContext = createContext({ mobile: false, displayCloseIcon: true } as {
  mobile: boolean;
  displayCloseIcon: boolean;
});

export const DrawerTitle: FC<HTMLAttributes<HTMLHeadingElement>> = ({ children, ...props }) => {
  const { mobile, displayCloseIcon } = useContext(DrawerContext);
  const asProp = mobile ? 'h6' : 'h5';
  return (
    <Title $mobile={mobile} $displayCloseIcon={displayCloseIcon} as={asProp} {...props}>
      {children}
    </Title>
  );
};

export const DrawerContent: FC<HTMLAttributes<HTMLDivElement>> = ({ children, ...props }) => {
  const contentRef = useRef<HTMLDivElement | null>(null);
  const mobile = useContext(DrawerContext).mobile;

  useLayoutEffect(() => {
    const node = contentRef.current;
    if (node) {
      resizePaddings(node);
      const resizeObserver = new ResizeObserver(debounce(() => resizePaddings(node), 250));
      resizeObserver.observe(node);
      return () => {
        resizeObserver.unobserve(node);
      };
    }
  }, []);

  return (
    <Content tabIndex={-1} ref={contentRef} $mobile={mobile} {...props}>
      {children}
    </Content>
  );
};

export const DrawerButtonPanel: FC<HTMLAttributes<HTMLDivElement>> = ({ children, ...props }) => {
  const mobile = useContext(DrawerContext).mobile;
  return (
    <ButtonPanel $mobile={mobile} {...props}>
      {children}
    </ButtonPanel>
  );
};
