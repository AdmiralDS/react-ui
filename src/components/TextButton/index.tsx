import { ButtonHTMLAttributes, ReactNode } from 'react';
import * as React from 'react';
import styled, { css, keyframes } from 'styled-components';
import { typography } from '#src/components/Typography';
import { Spinner } from '#src/components/Spinner';

type Dimension = 'm' | 's';
type Appearance = 'primary' | 'secondary';

const IconContainer = styled.div``;

const ButtonContainer = styled.div``;

const StyledSpinner = styled(Spinner)<{ dimension?: Dimension }>`
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  width: ${(p) => (p.dimension === 'm' ? '20px' : '16px')};
  height: ${(p) => (p.dimension === 'm' ? '20px' : '16px')};
`;

const loadingKeyframes = keyframes`
  to {
    background-position: 315px 0, 0 0, 0 190px, 50px 195px;
  }
`;

export const skeletonMixin = css`
  cursor: progress;
  background-image: linear-gradient(0.25turn, transparent, #fff, transparent), linear-gradient(#eee, #eee),
    radial-gradient(38px circle at 19px 19px, #eee 50%, transparent 51%), linear-gradient(#eee, #eee);
  background-repeat: no-repeat;
  background-size: 315px 250px, 315px 180px, 100px 100px, 225px 30px;
  background-position: -315px 0, 0 0, 0 190px, 50px 195px;
  animation: ${loadingKeyframes} 1.5s infinite;
`;

const StyledButton = styled.button<{
  dimension?: Dimension;
  appearance?: Appearance;
  displayRight?: boolean;
  $loading?: boolean;
  skeleton?: boolean;
}>`
  position: relative;
  box-sizing: border-box;
  display: inline-flex;
  flex-direction: ${(p) => (p.displayRight ? 'row-reverse' : 'row')};
  align-items: center;
  vertical-align: center;
  appearance: none;
  border: none;
  background: transparent;

  > ${IconContainer} {
    ${(p) => p.skeleton && skeletonMixin};
    border-radius: ${(p) => (p.skeleton ? '50%' : '')};
    svg {
      visibility: ${(p) => (p.skeleton || p.$loading ? 'hidden' : 'visible')};
      path {
        fill: ${(p) => (p.appearance === 'secondary' ? p.theme.color.text.primary : p.theme.color.basic.primary)};
      }
    }
  }
  > ${ButtonContainer} {
    ${(p) => p.skeleton && skeletonMixin};
    border-radius: ${(p) => (p.skeleton ? '4px' : '')};
    span {
      visibility: ${(p) => (p.skeleton || p.$loading ? 'hidden' : 'visible')};
    }
  }
  ${(p) => (p.dimension === 'm' ? typography['Button/Button 1'] : typography['Button/Button 2'])}

  color: ${(p) => (p.appearance === 'secondary' ? p.theme.color.text.primary : p.theme.color.basic.primary)};

  &:focus,
  &:hover {
    cursor: pointer;
    color: ${(p) => p.theme.color.basic.hover};
    > ${IconContainer} {
      svg {
        path {
          fill: ${(p) => p.theme.color.basic.hover};
        }
      }
    }
  }

  &:active {
    color: ${(p) => p.theme.color.basic.press};
    > ${IconContainer} {
      svg {
        path {
          fill: ${(p) => p.theme.color.basic.press};
        }
      }
    }
  }

  &:disabled {
    cursor: not-allowed;
    color: ${(p) => p.theme.color.text.tertiary};
  }
  > ${IconContainer} {
    width: ${(p) => (p.dimension === 'm' ? (p.skeleton ? '24px' : '20px') : p.skeleton ? '20px' : '16px')};
    height: ${(p) => (p.dimension === 'm' ? (p.skeleton ? '24px' : '20px') : p.skeleton ? '20px' : '16px')};
    margin: ${(p) => (p.displayRight ? '0 0 0 10px' : '0 10px 0 0')};
  }
`;

export interface TextButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  /** Размер кнопки */
  dimension?: Dimension;
  /** Внешний вид кнопки */
  appearance?: Appearance;
  /** Иконка кнопки */
  icon?: ReactNode;
  /** Текст кнопки */
  text?: string;
  /** Позиционирование иконки. По умолчанию - слева */
  displayRight?: boolean;
  /** Состояние загрузки */
  loading?: boolean;
  /** Состояние skeleton */
  skeleton?: boolean;
}

export const TextButton = React.forwardRef<HTMLButtonElement, TextButtonProps>(
  (
    {
      dimension = 'm',
      appearance = 'primary',
      icon,
      type = 'button',
      text,
      displayRight = false,
      loading = false,
      skeleton = false,
      ...props
    },
    ref,
  ) => {
    return (
      <StyledButton
        {...props}
        ref={ref}
        dimension={dimension}
        appearance={appearance}
        displayRight={displayRight}
        type={type}
        $loading={loading}
        skeleton={skeleton}
      >
        {loading && <StyledSpinner dimension={dimension} />}
        {icon && <IconContainer>{icon}</IconContainer>}
        <ButtonContainer>
          <span>{text}</span>
        </ButtonContainer>
      </StyledButton>
    );
  },
);

TextButton.displayName = 'TextButton';
