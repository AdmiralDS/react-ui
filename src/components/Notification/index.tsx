import type { HTMLAttributes } from 'react';
import * as React from 'react';
import styled, { css } from 'styled-components';
import { ReactComponent as InfoIcon } from '@admiral-ds/icons/build/service/InfoSolid.svg';
import { ReactComponent as WarningIcon } from '@admiral-ds/icons/build/service/ErrorSolid.svg';
import { ReactComponent as SuccessIcon } from '@admiral-ds/icons/build/service/CheckSolid.svg';
import { ReactComponent as ErrorIcon } from '@admiral-ds/icons/build/service/CloseSolid.svg';

import { LIGHT_THEME as DEFAULT_THEME } from '#src/components/themes';
import { typography } from '#src/components/Typography';
import { Link } from '#src/components/Link';
import { mediumGroupBorderRadius } from '#src/components/themes/borderRadius';
import { CloseIconPlacementButton } from '#src/components/IconPlacement';

export type NotificationStatus = 'info' | 'error' | 'success' | 'warning';

export interface NotificationProps extends Omit<HTMLAttributes<HTMLDivElement>, 'title' | 'id'> {
  /** Статус notification */
  status?: NotificationStatus;
  /** Заголовок notification */
  title?: React.ReactNode;
  /** Название для ссылки */
  linkText?: string;
  /** Url ссылки */
  href?: string;
  /** Переключатель видимости иконки "Close" */
  isClosable?: boolean;
  /** Закрытие notification */
  onClose?: () => void;
  /** Переключатель видимости статусных иконок */
  displayStatusIcon?: boolean;
}

const getIcon = (status: NotificationStatus) => {
  switch (status) {
    case 'info':
      return InfoIcon;
    case 'error':
      return ErrorIcon;
    case 'success':
      return SuccessIcon;
    case 'warning':
    default:
      return WarningIcon;
  }
};

const backGroundColorMixin = css<{ $status?: NotificationStatus }>`
  background-color: ${({ theme, $status }) => {
    switch ($status) {
      case 'warning':
        return `var(--admiral-color-Warning_Warning10, ${theme.color['Warning/Warning 10']})`;
      case 'error':
        return `var(--admiral-color-Error_Error10, ${theme.color['Error/Error 10']})`;
      case 'success':
        return `var(--admiral-color-Success_Success10, ${theme.color['Success/Success 10']})`;
      case 'info':
      default:
        return `var(--admiral-color-Primary_Primary10, ${theme.color['Primary/Primary 10']})`;
    }
  }};
`;

const borderColorMixin = css<{ $status?: NotificationStatus }>`
  border-color: ${({ theme, $status }) => {
    switch ($status) {
      case 'warning':
        return `var(--admiral-color-Warning_Warning50Main, ${theme.color['Warning/Warning 50 Main']})`;
      case 'error':
        return `var(--admiral-color-Error_Error60Main, ${theme.color['Error/Error 60 Main']})`;
      case 'success':
        return `var(--admiral-color-Success_Success50Main, ${theme.color['Success/Success 50 Main']})`;
      case 'info':
      default:
        return `var(--admiral-color-Primary_Primary60Main, ${theme.color['Primary/Primary 60 Main']})`;
    }
  }};
`;

const NotificationWrapper = styled.div<{
  $status?: NotificationStatus;
  $displayStatusIcon: boolean;
  $isClosable: boolean;
}>`
  overflow: hidden;
  position: relative;
  box-sizing: border-box;
  border-radius: var(--admiral-border-radius-Medium, ${(p) => mediumGroupBorderRadius(p.theme.shape)});
  padding: 12px 44px 12px 52px;
  ${(p) => (p.$displayStatusIcon ? '' : 'padding-left: 16px;')}
  ${(p) => (p.$isClosable ? '' : 'padding-right: 16px;')}
  border-width: 1px;
  border-style: solid;
  ${borderColorMixin};
  ${backGroundColorMixin};
`;

const Content = styled.div`
  display: flex;
  flex-direction: column;
  padding: 2px 0;
`;

const Title = styled.div`
  ${typography['Subtitle/Subtitle 3']}
  color: var(--admiral-color-Neutral_Neutral90, ${(p) => p.theme.color['Neutral/Neutral 90']});
  margin-bottom: 4px;
`;

const CustomBody = styled.div`
  ${typography['Body/Body 2 Long']}
  color: var(--admiral-color-Neutral_Neutral90, ${(p) => p.theme.color['Neutral/Neutral 90']});
`;

const CloseButton = styled(CloseIconPlacementButton)`
  position: absolute;
  top: 10px;
  right: 8px;
`;

const IconWrapper = styled.div<{ $status?: NotificationStatus }>`
  position: absolute;
  top: 12px;
  left: 16px;
  width: 24px;
  height: 24px;
  & svg {
    width: 24px;
    height: 24px;
    & *[fill^='#'] {
      fill: ${({ theme, $status }) => {
        switch ($status) {
          case 'warning':
            return `var(--admiral-color-Warning_Warning50Main, ${theme.color['Warning/Warning 50 Main']})`;
          case 'error':
            return `var(--admiral-color-Error_Error60Main, ${theme.color['Error/Error 60 Main']})`;
          case 'success':
            return `var(--admiral-color-Success_Success50Main, ${theme.color['Success/Success 50 Main']})`;
          case 'info':
          default:
            return `var(--admiral-color-Primary_Primary60Main, ${theme.color['Primary/Primary 60 Main']})`;
        }
      }};
    }
  }
`;

const LinkWrapper = styled(Link)`
  margin-top: 4px;
  ${typography['Button/Button 2']}
`;

NotificationWrapper.defaultProps = {
  theme: DEFAULT_THEME,
};

/**
 * @deprecated use NotificationItem component instead
 */
export const Notification = ({
  status = 'info',
  title,
  linkText = '',
  href = '',
  displayStatusIcon = false,
  isClosable = false,
  onClose,
  children,
  ...props
}: React.PropsWithChildren<NotificationProps>) => {
  const NotificationIcon = getIcon(status);

  const isAlert = status !== 'info';

  return (
    <NotificationWrapper
      role={isAlert ? 'alert' : 'status'}
      aria-live={isAlert ? 'assertive' : 'polite'}
      $status={status}
      $displayStatusIcon={displayStatusIcon}
      $isClosable={isClosable}
      {...props}
    >
      {displayStatusIcon && (
        <IconWrapper $status={status}>
          <NotificationIcon />
        </IconWrapper>
      )}
      <Content>
        {title && <Title>{title}</Title>}
        <CustomBody>{children}</CustomBody>
        {linkText && <LinkWrapper href={href}>{linkText || href}</LinkWrapper>}
      </Content>

      {isClosable && <CloseButton dimension="mSmall" highlightFocus={false} onClick={onClose} />}
    </NotificationWrapper>
  );
};

Notification.displayName = 'Notification';
