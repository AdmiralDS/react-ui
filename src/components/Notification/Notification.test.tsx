import * as React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { LIGHT_THEME } from '#src/components/themes';
import {
  Notification,
  NotificationButtonPanel,
  NotificationContent,
  NotificationTitle,
} from '#src/components/Notification';
import { TextButton } from '#src/components/TextButton';

describe('Notification', () => {
  it('should render error components with all properties', () => {
    const handleClick = jest.fn();
    const wrapper = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <Notification status="info" isClosable displayStatusIcon onClose={handleClick}>
          text
          <NotificationTitle>Title</NotificationTitle>
          <NotificationContent>text</NotificationContent>
          <NotificationButtonPanel>
            <TextButton dimension="s" text="Link" />
          </NotificationButtonPanel>
        </Notification>
      </ThemeProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
