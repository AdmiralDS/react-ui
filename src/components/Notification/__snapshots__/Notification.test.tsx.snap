// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification should render error components with all properties 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c5 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  pointer-events: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  color: #0D69F2;
  font-size: 16px;
  font-weight: normal;
  line-height: 24px;
  font-family: 'VTB Group UI',sans-serif;
}

.c5:hover {
  color: #0A54C2;
}

.c5:hover *[fill^='#'] {
  fill: #0A54C2;
}

.c5:focus {
  color: #0A54C2;
  outline-color: #0A54C2;
}

.c5:focus *[fill^='#'] {
  fill: #0A54C2;
}

.c5:active {
  color: #0949AA;
}

.c5:active *[fill^='#'] {
  fill: #0949AA;
}

.c5 *[fill^='#'] {
  fill: #0D69F2;
}

.c0 {
  overflow: hidden;
  position: relative;
  box-sizing: border-box;
  border-radius: 4px;
  padding: 12px 44px 12px 52px;
  border-width: 1px;
  border-style: solid;
  border-color: #0D69F2;
  background-color: #F1F2F4;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 2px 0px;
}

.c3 {
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
  color: #2B313B;
  margin-bottom: 4px;
}

.c4 {
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
  color: #2B313B;
}

.c7 {
  position: absolute;
  top: 14px;
  right: 12px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 20px;
  height: 20px;
}

.c7 svg {
  width: 20px;
  height: 20px;
}

.c7 svg *[fill^='#'] {
  fill: #626F84;
}

.c7 svg:hover *[fill^='#'] {
  fill: #0A54C2;
}

.c1 {
  position: absolute;
  top: 11px;
  left: 15px;
  width: 24px;
  height: 24px;
}

.c1 svg {
  width: 24px;
  height: 24px;
}

.c1 svg *[fill^='#'] {
  fill: #0D69F2;
}

.c6 {
  margin-top: 4px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
  color: #0D69F2;
}

<body>
    <div>
      <div
        aria-live="polite"
        class="c0"
        role="status"
      >
        <div
          class="c1"
        >
          <div />
        </div>
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            Title
          </div>
          <div
            class="c4"
          >
            text
          </div>
          <a
            class="c5 c6"
            href="http://localhost"
          >
            Link
          </a>
        </div>
        <div
          class="c7"
          tabindex="0"
        >
          <div />
        </div>
      </div>
    </div>
  </body>,
  "container": .c5 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  pointer-events: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  color: #0D69F2;
  font-size: 16px;
  font-weight: normal;
  line-height: 24px;
  font-family: 'VTB Group UI',sans-serif;
}

.c5:hover {
  color: #0A54C2;
}

.c5:hover *[fill^='#'] {
  fill: #0A54C2;
}

.c5:focus {
  color: #0A54C2;
  outline-color: #0A54C2;
}

.c5:focus *[fill^='#'] {
  fill: #0A54C2;
}

.c5:active {
  color: #0949AA;
}

.c5:active *[fill^='#'] {
  fill: #0949AA;
}

.c5 *[fill^='#'] {
  fill: #0D69F2;
}

.c0 {
  overflow: hidden;
  position: relative;
  box-sizing: border-box;
  border-radius: 4px;
  padding: 12px 44px 12px 52px;
  border-width: 1px;
  border-style: solid;
  border-color: #0D69F2;
  background-color: #F1F2F4;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 2px 0px;
}

.c3 {
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
  color: #2B313B;
  margin-bottom: 4px;
}

.c4 {
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
  color: #2B313B;
}

.c7 {
  position: absolute;
  top: 14px;
  right: 12px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 20px;
  height: 20px;
}

.c7 svg {
  width: 20px;
  height: 20px;
}

.c7 svg *[fill^='#'] {
  fill: #626F84;
}

.c7 svg:hover *[fill^='#'] {
  fill: #0A54C2;
}

.c1 {
  position: absolute;
  top: 11px;
  left: 15px;
  width: 24px;
  height: 24px;
}

.c1 svg {
  width: 24px;
  height: 24px;
}

.c1 svg *[fill^='#'] {
  fill: #0D69F2;
}

.c6 {
  margin-top: 4px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
  color: #0D69F2;
}

<div>
    <div
      aria-live="polite"
      class="c0"
      role="status"
    >
      <div
        class="c1"
      >
        <div />
      </div>
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          Title
        </div>
        <div
          class="c4"
        >
          text
        </div>
        <a
          class="c5 c6"
          href="http://localhost"
        >
          Link
        </a>
      </div>
      <div
        class="c7"
        tabindex="0"
      >
        <div />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
